jetty-server/src/main/java/org/eclipse/jetty/server/handler/AllowSymLinkAliasChecker.java

```diff
@@ -99,4 +88,22 @@ public boolean check(String uri, Resource resource)
         return false;
     }
 
+    private Path unrollSymbolicLink(Path path) throws IOException
+    {
+      Path d = path.getRoot();
+      for (Path e:path)
+      {
+          d=d.resolve(e);
+
+          while (Files.exists(d) && Files.isSymbolicLink(d))
+          {
+              Path link=Files.readSymbolicLink(d);
+              if (!link.isAbsolute())
+                  link=d.resolve(link);
+              d=unrollSymbolicLink(link);
```

Comment:
I'm still puzzled exactly why recursion is needed.... if we just had d=link, wouldn't it be unrolled by the while loop?  What am I missing???